### PR TITLE
Remove Flow types from Scheduler and Tracing

### PR DESCRIPTION
**Pull request by sebmarkbage**
The idea is to make the scheduler as unopinionated and uncontroversial as possible.

Currently it's a bit of a mix and match and also use our build tooling. This keeps spreading.

Let's make a decision here either way and stick to it.

### CODE REVIEW

File: scripts/flow/environment.js [hunk 1]
```diff
@@ -41,3 +41,78 @@ declare module 'EventListener' {
     capture: (target: Element, type: string, callback: Function) => mixed,
   };
 }
+
+// Libdefs for Scheduler
+
+declare module 'scheduler/tracing' {
```
Comment (sebmarkbage) [inline scripts/flow/environment.js]: This can go into flowtyped.

**Comment by bvaughn** [comment]
Mixed feels about this.

**Comment by acdlite** [comment]
I thought we agreed we’d make this decision after the API stabilizes and if/when we’re sure we want to push other frameworks (outside React ecosystem) to adopt it. Both are still up in the air.

**Comment by bvaughn** [comment]
Agreed. Feels like removing the types now will just make the next few iterations a little more painful.

**Comment by Usamaliaquat123** [comment]
@gaearon  please tell me how can i start contributing in react libr any docs thanks man

File: scripts/flow/environment.js [hunk 2]
```diff
@@ -41,3 +41,78 @@ declare module 'EventListener' {
...
+
+  declare function unstable_getCurrent(): Set<Interaction> | null;
+
+  declare function unstable_getThreadID(): number;
+
+  declare function unstable_trace(
+    name: string,
+    timestamp: number,
+    callback: Function,
+    threadID: number,
```
Comment (eps1lon) [inline scripts/flow/environment.js]: This argument is optional in the implementation  https://github.com/facebook/react/blob/63f9f60376816bde524dcd15f0094cb8eb49188a/packages/scheduler/src/Tracing.js#L64 and in `unstable_wrap`. Not sure which one is right.

File: scripts/flow/environment.js [hunk 3]
```diff
@@ -41,3 +41,78 @@ declare module 'EventListener' {
...
+    // In that case, onWorkStarted may be called more than once before onWorkStopped.
+    // Work is done by a "thread" which is identified by a unique ID.
+    onWorkStarted: (interactions: Set<Interaction>, threadID: number) => void,
+
+    // A batch of work has completed for a set of interactions.
+    // Work is done by a "thread" which is identified by a unique ID.
+    onWorkStopped: (interactions: Set<Interaction>, threadID: number) => void,
+  };
+
+  declare type InteractionsRef = {
```
Comment (eps1lon) [inline scripts/flow/environment.js]: This is typed like `enableSchedulerTracing` is `true` but `  declare function unstable_getCurrent(): Set<Interaction> | null;` is typed as `enableSchedulerTracing` is either `true` or `false`.

Should these types cover all builds of `scheduler/tracing` or only `tracing.development`. If it should cover all builds this should be nullable.

**Comment by gaearon** [comment]
Stale.

**Pull request closed without merging**